### PR TITLE
Remove untrusted workload annotation support.

### LINKED ISSUE #1095: Should we remove trusted/untrusted runtime references from the code?
The `RuntimeClass` is now a well known and supported feature in k8s, hence I was wondering if we could simplify the code picking the right container runtime by removing the references to `trusted`/`untrusted` (mainly [here](https://github.com/containerd/cri/blob/master/pkg/server/sandbox_run.go)).

### PR DESCRIPTION
**Pull request by Random-Liu**
Fixes https://github.com/containerd/cri/issues/1095
For https://github.com/containerd/cri/issues/1257

I sent this out because of `// TODO(windows): Deprecate this so that we don't need to handle it for windows`.

The deprecation justification:
* We announced deprecation of these annotations in the containerd v1.2.0 [release note](https://github.com/containerd/containerd/releases/tag/v1.2.0):
```
The plugins.cri.containerd.untrusted_workload_runtime config option and io.kubernetes.cri.untrusted-workload pod annotation are still functional, but start being deprecated. It is recommended to migrate to the RuntimeClass api.
```
Based on our [configuration deprecation policy](https://github.com/containerd/cri/blob/master/docs/config.md#deprecation), a deprecated option `is kept functional for 6 months or 1 release (whichever is longer)`. We've kept it for **1 year, one 1.3 release**.
* The annotation based feature is treated as experimental since the beginning.
* Kubernetes 1.14 has runtime class beta enabled by default. Kubernetes 1.16 is almost out. Containerd 1.4 will likely be released after Kubernetes 1.17, at which time supported Kubernetes versions in upstream will be 1.14, 1.15, 1.16, 1.17.

Of course we should include the removal of the support into containerd 1.4 release note.

Signed-off-by: Lantao Liu <lantaol@google.com>

### CODE REVIEW

**Review by jterry75 — APPROVED** [review]
LGTM

**Review by AkihiroSuda — APPROVED** [review]

File: docs/config.md [hunk 1]
```diff
@@ -90,16 +90,6 @@ version = 2
...
-    # 'plugins."io.containerd.grpc.v1.cri".containerd.default_runtime' is the runtime to use in containerd.
-    # DEPRECATED: use `default_runtime_name` and `plugins."io.containerd.grpc.v1.cri".runtimes` instead.
-    # Remove in containerd 1.4.
-    [plugins."io.containerd.grpc.v1.cri".containerd.default_runtime]
-
-    # 'plugins."io.containerd.grpc.v1.cri".containerd.untrusted_workload_runtime' is a runtime to run untrusted workloads on it.
-    # DEPRECATED: use `untrusted` runtime in `plugins."io.containerd.grpc.v1.cri".runtimes` instead.
-    # Remove in containerd 1.4.
-    [plugins."io.containerd.grpc.v1.cri".containerd.untrusted_workload_runtime]
-
```
Comment (mikebrow) [inline docs/config.md]: removing from here is fine... but the significant number of deletions do bring up a thought that maybe we need a doc or appendix for admins to help them transition from the deleted config items.

**Review by mikebrow — APPROVED** [review]
LGTM

but see comment

**Comment by AkihiroSuda** [comment]
needs rebase

**Comment by mikebrow** [comment]
needs a rebase @Random-Liu

**Comment by mikebrow** [comment]
@Random-Liu

**Pull request closed without merging**